source/server/BUILD

```diff
@@ -208,8 +208,6 @@ envoy_cc_library(
         "//source/common/network:utility_lib",
         "//source/common/protobuf:utility_lib",
         "//source/common/ssl:context_config_lib",
-        "//source/server/config/listener:original_dst_lib",
-        "//source/server/config/listener:proxy_protocol_lib",
```

Comment:
I think this is wrong, both of those filters can be implicitly injected if config uses [`use_original_dst`](https://github.com/envoyproxy/envoy/blob/1260bf1c1f6617f19bea212cbe75ee49842d1b8a/source/server/listener_manager_impl.cc#L150) or [`use_proxy_protocol`](https://github.com/envoyproxy/envoy/blob/1260bf1c1f6617f19bea212cbe75ee49842d1b8a/source/server/listener_manager_impl.cc#L161), and while this doesn't break the binary, because those filters are loaded as extensions, it breaks the dependency tree.

cc @mattklein123 @htuch

Reply:
https://github.com/envoyproxy/envoy/pull/2947 re-adds them.

Reply:
Doesn't https://github.com/envoyproxy/envoy/pull/2944/files#diff-93760bebb8e9ba9b0c9846c9412f75dfR9 cover this? I.e. we now load these in the same way as other filters.

Reply:
It links those (and possibly all other dynamic extensions) into final binary, yes.

However, those 2 filters are used directly in `ListenerManager` ([here](https://github.com/envoyproxy/envoy/blob/1260bf1c1f6617f19bea212cbe75ee49842d1b8a/source/server/listener_manager_impl.cc#L150) and [here](https://github.com/envoyproxy/envoy/blob/1260bf1c1f6617f19bea212cbe75ee49842d1b8a/source/server/listener_manager_impl.cc#L161)), so this class directly depends on them, whereas other extensions inject themselves at runtime and are only bundled together via config.

Reply:
It depends on them still only at runtime, there is no header dependency (which to me is the sniff test for BUILD deps). It's totally reasonable to take a `listener_manager_lib`, link it into some other application that doesn't make use of these listener filters in its config and not have to force a dependency on their build.